modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/StepFilterProvider.scala

```diff
@@ -9,13 +9,21 @@ import ch.epfl.scala.debugadapter.internal.stepfilter._
 import ch.epfl.scala.debugadapter.Debuggee
 import ch.epfl.scala.debugadapter.DebugTools
 import com.sun.jdi.Location
+import java.util.Optional
 
 class StepFilterProvider(
     stepFilters: Seq[StepFilter],
     logger: Logger,
     testMode: Boolean
 ) extends JavaStepFilterProvider() {
 
+  override def shouldSkipFrame(method: Method): Boolean = {
```

Comment:
We should probably rename the class. It is not only about skipping frame but also about formatting them. 

What about `StackTraceProvider`?

Reply:
Do we still need this method? Can we use the formatMethodName instead, since it returns an Optional?